mlos_core/mlos_core/optimizers/bayesian_optimizers/smac_optimizer.py

```diff
@@ -365,3 +517,41 @@ def _to_configspace_configs(self, *, configs: pd.DataFrame) -> List[ConfigSpace.
             ConfigSpace.Configuration(self.optimizer_parameter_space, values=config.to_dict())
             for (_, config) in configs.astype('O').iterrows()
         ]
+
+    def _extract_config(self, trial: TrialInfo) -> pd.DataFrame:
+        """Convert TrialInfo to a config DataFrame.
+
+        Parameters
+        ----------
+        trial : TrialInfo
+            The trial to extract.
+
+        Returns
+        -------
+        config : pd.DataFrame
+            Pandas dataframe with a single row containing the config.
+            Column names are config parameters
+        """
+        return pd.DataFrame(
+            [trial.config], columns=list(self.optimizer_parameter_space.keys())
+        )
+
+
+def _extract_metadata(trial: TrialInfo) -> pd.DataFrame:
+    """Convert TrialInfo to a metadata DataFrame.
+
+    Parameters
+    ----------
+    trial : TrialInfo
+        The trial to extract.
+
+    Returns
+    -------
+    metadata : pd.DataFrame
+        Pandas dataframe with a single row containing the metadata.
+        Column names are the budget and instance of the evaluation, if valid.
+    """
+    return pd.DataFrame(
+        [[trial.instance, trial.seed, trial.budget]],
+        columns=["instance", "seed", "budget"],
```

Comment:
Comments about why these columns in particular?

Reply:
These are the only ones that exist on the object. Added a comment